src/NzbDrone.Api/Calendar/CalendarFeedModule.cs

```diff
@@ -101,12 +109,23 @@ private Response GetCalendarFeed()
 
                 var occurrence = calendar.Create<Event>();
                 occurrence.Uid = "NzbDrone_episode_" + episode.Id;
-                occurrence.Status = episode.HasFile ? EventStatus.Confirmed : EventStatus.Tentative;
-                occurrence.Start = new CalDateTime(episode.AirDateUtc.Value) { HasTime = true };
-                occurrence.End = new CalDateTime(episode.AirDateUtc.Value.AddMinutes(episode.Series.Runtime)) { HasTime = true };
-                occurrence.Description = episode.Overview;
+				occurrence.Status = episode.HasFile ? EventStatus.Confirmed : EventStatus.Tentative;
+				occurrence.Description = episode.Overview;
                 occurrence.Categories = new List<string>() { episode.Series.Network };
 
+				if (allDay)
+				{
+					DateTime startDate = episode.AirDateUtc.Value;
+					DateTime endDate = episode.AirDateUtc.Value.AddDays(1);
+					occurrence.Start = new CalDateTime(startDate.Year, startDate.Month, startDate.Day) { HasTime = false };
+					occurrence.End = new CalDateTime(endDate.Year, endDate.Month, endDate.Day) { HasTime = false };
```

Comment:
Based on webcal spec you can just leave `.End` unset if you have `.Start` without time, can you try that out? Saves you the arithmetic, and other calendar apps may freak out with the end date turning up as the next day... or the next year if it happens to air on December 31st.